### DISCUSSION TITLE
Only returns the fields that contains a match

### DISCUSSION THREAD

**Discussion by kiastorm** [meilisearch/product#782] [Feedback & Feature Proposal]
I have a situation where I have an array field, `Content`, that I'm searching for a search term.

example:
```
[
  {
    Content: [
      "hello there",
      "testing",
      "want to say hello"
    ]  
  },
  {
    Content: [
      "some other string"
    ]
  }
]
```

If my search term is `hello`, I want this to return

```
[
  {
    Content: [
      "<mark>hello</mark> there",
      "want to say <mark>hello</mark>"
    ]  
  },
]
```

Essentially, I want only the array items that actually match to be returned

Currently, I'm filtering through on the client side, but this invalidates the `totalResults` and `totalPages`

```
// Leave out any pages that do not contain highlighted matches
for (const hit of searchResults.hits) {
  const formattedHit = {
    ...hit,
    _formatted: {
      ...hit._formatted,
      Content: [],
    },
  };

  for (const page of hit._formatted.Content) {
    if (page.includes("<mark>")) {
      formattedHit._formatted.Content.push(page);
    }
  }

  if (formattedHit._formatted.Content.length) {
    formattedHits.push(formattedHit);
  }
}
```

Is this possible already? Any suggestions?

**Comment by irevoire** [comment]
Hey,

No, this isn’t possible currently. I'm sorry. You can only specify in advance which fields you want and can’t select them dynamically while returning hits.
Your code is pretty good; I don’t think you can improve it much.

I’ll open a product request for you so we can track this feature request on our side; thanks for the feedback!
